### PR TITLE
Add tests to verify the correctness of index in distributed mode

### PR DESCRIPTION
**Pull request by hitchhicker**

### CODE REVIEW

File: tests/unit/test_quantize.py [hunk 1]
```diff
@@ -225,3 +226,68 @@ def test_quantize_with_empty_and_non_empty_files(tmpdir):
...
+    )
+    temporary_indices_folder = os.path.join(tmpdir.strpath, "distributed_autofaiss_indices")
+    ids_path = os.path.join(tmpdir.strpath, "ids")
+    index, _ = build_index(
+        embeddings=tmp_dir,
+        distributed="pyspark",
+        file_format="parquet",
+        temporary_indices_folder=temporary_indices_folder,
+        max_index_memory_usage="100MB",
+        current_memory_available="120MB",
```
Comment (hitchhicker) [inline tests/unit/test_quantize.py:249]: This needs to adapt to the new change for the memory that is done in the branch `n-indices`

File: tests/unit/test_quantize.py [hunk 2]
```diff
@@ -225,3 +226,68 @@ def test_quantize_with_empty_and_non_empty_files(tmpdir):
...
+        tmpdir, min_size=min_size, max_size=max_size, dim=dim, nb_files=nb_files
+    )
+
+    index, _ = build_index(
+        embeddings=tmp_dir,
+        distributed="pyspark",
+        file_format="npy",
+        temporary_indices_folder=temporary_indices_folder,
+        max_index_memory_usage="100MB",
+        current_memory_available="120MB",
```
Comment (hitchhicker) [inline tests/unit/test_quantize.py:278]: Same here

File: tests/unit/test_quantize.py [hunk 3]
```diff
@@ -225,3 +226,68 @@ def test_quantize_with_empty_and_non_empty_files(tmpdir):
...
+
+    query = faiss.rand((1, dim))
+    distances, ids = index.search(query, k=9)
+
+    ground_truth_index = faiss.index_factory(dim, "IVF1,Flat")
+    ground_truth_index.train(expected_array)
+    ground_truth_index.add(expected_array)
+    ground_truth_distances, ground_truth_ids = ground_truth_index.search(query, k=9)
+
+    assert_array_equal(ids, ground_truth_ids)
```
Comment (rom1504) [inline tests/unit/test_quantize.py:293]: could you also check the mapping ids and not only the consecutive ids ?

Comment (hitchhicker) [reply tests/unit/test_quantize.py:297]: done now.

**Pull request merged**